libs/vkd3d/swapchain.c

```diff
@@ -1358,12 +1358,13 @@ static HRESULT d3d12_swapchain_create_vulkan_swapchain(struct d3d12_swapchain *s
         return hresult_from_vk_result(vr);
     }
 
-    image_count = swapchain->desc.BufferCount;
+    /* Need to account for the front buffer, so + 1 */
+    image_count = swapchain->desc.BufferCount + 1;
     image_count = max(image_count, surface_caps.minImageCount);
     if (surface_caps.maxImageCount)
         image_count = min(image_count, surface_caps.maxImageCount);
 
-    if (image_count != swapchain->desc.BufferCount)
+    if (image_count != (swapchain->desc.BufferCount + 1))
```

Comment:
Do we need this WARN at all?

IMO we should just do our thing here and stay quiet since the app never sees the effect of having too many Vulkan back buffers anyway.

Reply:
We don't now, since we always use userbuffers.